app/job/rest.py

```diff
@@ -53,6 +55,15 @@ def get_job_by_service_and_job_id(service_id, job_id):
     return jsonify(data=data)
 
 
+@job.route('/<job_id>/cancel', methods=['POST'])
+def cancel_job(service_id, job_id):
+    job = dao_get_future_scheduled_job_by_id_and_service_id(job_id, service_id)
+    job.job_status = JOB_STATUS_CANCELLED
+    dao_update_job(job)
+
+    return get_job_by_service_and_job_id(service_id, job_id)
```

Comment:
I’m calling the method for getting a job because it saves repeating:
- adding the statistics to the job
- JSONifying the response

See: https://github.com/alphagov/notifications-api/blob/add-cancel-job-endpoint/app/job/rest.py#L50-L55